UVAtlas/inc/JntLinuxTypes.h

```diff
@@ -0,0 +1,1653 @@
+#pragma once
+
+#if defined(WIN32)
+#define _CDECL __cdecl
+#elif defined(__LINUX__) || defined(_PSY_LINUX_)
+#define _CDECL __attribute__((__cdecl__))
+#endif
+
+#if defined(__LINUX__) || defined(_PSY_LINUX_)
+
+#include <dxgiformat.h>
+#include <wchar.h>
+#include <cstdarg>
+
+//Sal annotations
+#define _Use_decl_annotations_
+#define _In_
+#define _In_opt_
+#define _In_z_
+#define _In_reads_(size)
+#define _In_reads_opt_(size)
+#define _In_reads_bytes_(size)
+#define _Out_
+#define _Out_opt_
+#define _Out_writes_(size)
+#define _Out_writes_opt_(size)
+#define _Out_writes_bytes_(size)
+#define _Inout_
+#define _Inout_opt_
+#define _Inout_updates_all_(size)
+#define _Inout_updates_all_opt_(size)
+#define _Inout_updates_bytes_all_(size)
+#define _When_(expr, annotes)
+#define _Printf_format_string_
+#define _Analysis_assume_(expr)
+#define __in_ecount(size)
+#define __out_ecount(size)
+#define _Success_(expr)
+#define _Analysis_assume_(expr)
+
+//------------------------------------------------------------------------------
+
+#ifndef _HRESULT_DEFINED
+#define _HRESULT_DEFINED
+typedef long HRESULT;
+#endif // !_HRESULT_DEFINED
+
+//HRESULT Codes
+#define S_OK                             ((HRESULT)0L)
+#define S_FALSE                          ((HRESULT)1L)
+#define E_FAIL                           ((HRESULT)0x80004005L)
+#define E_OUTOFMEMORY                    ((HRESULT)0x8007000EL)
+#define E_INVALIDARG                     ((HRESULT)0x80070057L)
+#define E_UNEXPECTED                     ((HRESULT)0x8000FFFFL)
+#define E_ABORT                          ((HRESULT)0x80004004L)
+#define E_POINTER                        ((HRESULT)0x80004003L)
+#define ERROR_INVALID_DATA               13L
+#define ERROR_ARITHMETIC_OVERFLOW        534L
+#define ERROR_NOT_SUPPORTED              50L
+
+#define SUCCEEDED(hr) (((HRESULT)(hr)) >= 0)
+#define FAILED(hr) (((HRESULT)(hr)) < 0)
+#define HRESULT_FROM_WIN32(x) (x)
+
+//------------------------------------------------------------------------------
+
+#define WINAPI __attribute__((stdcall))
+#define STDMETHOD(method) virtual __attribute__((nothrow,stdcall)) HRESULT method
+#define STDMETHODIMP HRESULT __attribute__((stdcall))
+#define PURE = 0
+#define UNREFERENCED_PARAMETER(P) (P)
+
+#define __max(a,b) (((a) > (b)) ? (a) : (b))
+#define __min(a,b) (((a) < (b)) ? (a) : (b))
+static int swprintf_s(wchar_t* buff, const wchar_t* fmt, ...)
+{
+    va_list args;
+    va_start(args, fmt);
+    vswprintf(buff, sizeof(buff) / sizeof(*buff), fmt, args);
+};
+
+typedef unsigned long DWORD;
+typedef const char* LPCSTR;
+typedef void* HANDLE;
+typedef long long LONG_PTR;
+#define INVALID_HANDLE_VALUE ((HANDLE)(LONG_PTR)-1)
+
+//------------------------------------------------------------------------------
+
+//DirectXMath Port
+
+#define XM_CONST constexpr
+#define XM_CONSTEXPR constexpr
+#define XM_CALLCONV __attribute__((fastcall))
+
+namespace DirectX
+{
+
+XM_CONST float XM_PI        = 3.141592654f;
```

Comment:
may want to define these to whats included in <math.h>

Reply:
My preference here is to stick with the same values as the MS SDK. This was a copy paste.